packages/gestalt/src/TileData.js

```diff
@@ -0,0 +1,149 @@
+// @flow strict
+import { type Node } from 'react';
+import { type Indexable } from './zIndex.js';
+import { useColorScheme } from './contexts/ColorSchemeProvider.js';
+import InternalDatapoint from './Datapoint/InternalDatapoint.js';
+import Tile from './Tile/Tile.js';
+
+type TooltipProps = {|
+  accessibilityLabel?: string,
+  inline?: boolean,
+  idealDirection?: 'up' | 'right' | 'down' | 'left',
+  text: string,
+  zIndex?: Indexable,
+|};
+
+export type TileChangeHandler = ({|
+  event:
+    | SyntheticMouseEvent<HTMLDivElement>
+    | SyntheticKeyboardEvent<HTMLDivElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  selected: boolean,
+  id?: string,
+|}) => void;
+
+// Note: 03 is ommited because it doesn't have a corresponding dark token
+type DataVisualizationColors =
+  | '01'
+  | '02'
+  | '04'
+  | '05'
+  | '06'
+  | '07'
+  | '08'
+  | '09'
+  | '10'
+  | '11'
+  | '12';
+
+type TrendObject = {|
+  accessibilityLabel: string,
+  value: number,
+|};
+
+type Props = {|
+  /**
+   * A valid color code from the [data visualization palette](https://gestalt.pinterest.systems/foundations/data_visualization/palette). 03 is not supported yet, because it has no corresponding dark theme.
+   */
+  color?: DataVisualizationColors,
+  /**
+   * Indicates if TileData should be disabled. Disabled TileData is inactive and cannot be interacted with.
+   */
+  disabled?: boolean,
+  /**
+   * An optional identifier to be passed back in the onChange callback. It can be helpful to distinguish multiple DataPoints.
+   */
+  id?: string,
+  /**
+   * Handler if the item selection state is changed.
+   */
+  onChange?: TileChangeHandler,
+  /**
+   * Controls whether the tile is selected or not.
+   */
+  selected?: boolean,
+  /**
+   * Shows a visible checkbox when the tile is selected state. See when using in a [group](https://gestalt.pinterest.systems/web/tiledata#Group).
+   */
+  showCheckbox?: boolean,
```

Comment:
Is this prop necessary? When would we want to allow for selection and not show a checkbox? What happens if `showCheckbox` is true but neither `selected` nor `onChange` are present?

Reply:
I understand the "selection without a checkbox" use case, but what happens if the user passes `showCheckbox` and neither `selected` nor `onTap`?